search/search_quality/search_quality_tests/real_mwm_tests.cpp

```diff
@@ -326,4 +353,100 @@ UNIT_CLASS_TEST(MwmTestsFixture, Germany_Karlstraße_3)
 }
+
+// https://github.com/organicmaps/organicmaps/issues/3318
+// https://github.com/organicmaps/organicmaps/issues/3317
+UNIT_CLASS_TEST(MwmTestsFixture, IceCream)
+{
+  // Load all USA.
+  RegisterLocalMapsByPrefix("US_");
+  // Hilo, Hawaii
+  ms::LatLon const center{19.7073734, -155.0815800};
+  SetViewport(center, 3000);
+
+  auto request = MakeRequest("Gelato");
+  auto const & results = request->Results();
+  size_t constexpr kResultsCount = 10;
+  TEST_GREATER(results.size(), kResultsCount, ());
+
+  Range const range(results, 0, kResultsCount);
+  EqualClassifType(range, GetClassifTypes({{"amenity", "ice_cream"}, {"cuisine", "ice_cream"}}));
+  TEST_LESS(SortedByDistance(range, center), 2000.0, ());
+
+  auto request2 = MakeRequest("Ice cream");
+  auto const & results2 = request2->Results();
+  TEST_GREATER(results2.size(), kResultsCount, ());
+
+  for (size_t i = 0; i < kResultsCount; ++i)
+    TEST(results[i].GetFeatureID() == results2[i].GetFeatureID(), (results[i], results2[i]));
+}
+
+UNIT_CLASS_TEST(MwmTestsFixture, Hilo_City)
+{
+  // Istanbul, Kadikoy.
+  ms::LatLon const center(40.98647, 29.02552);
+  SetViewportAndLoadMaps(center);
+
+  // Lets start with trailing space here.
+  // Prefix search is more fuzzy and gives "Hill", "Holo", .. nearby variants.
+  auto request = MakeRequest("Hilo ");
+  auto const & results = request->Results();
+  size_t constexpr kResultsCount = 5;  // Hilo city in Hawaii should be at the top.
+  TEST_GREATER(results.size(), kResultsCount, ());
+
+  auto const cityType = classif().GetTypeByPath({"place", "city"});
+
+  bool found = false;
+  for (size_t i = 0; i < kResultsCount; ++i)
+  {
+    auto const & r = results[i];
+    if (r.GetResultType() == search::Result::Type::Feature &&
+        r.GetString() == "Hilo" &&
+        EqualClassifType(r.GetFeatureType(), cityType))
+    {
+      found = true;
+      break;
+    }
+  }
+
+  TEST(found, (results));
+}
+
+// https://github.com/organicmaps/organicmaps/issues/3266
+// Moscow has suburb "Арбат" and many streets, starting from number (2-й Обыденский), (4-й Голутвинский) inside.
+// So "Арбат 2" is a _well-ranked_ street result like [Suburb full match, Street full prefix],
+// but we obviously expect the building "улица Арбат 2 с/1", which is a _low-ranked_ substring.
+UNIT_CLASS_TEST(MwmTestsFixture, Arbat_Address)
+{
+  // Moscow, Arbat
+  ms::LatLon const center(55.74988, 37.59240);
+  SetViewportAndLoadMaps(center);
+
+  for (auto const & query : {"Арбат 2", "Арбат 4"})
+  {
+    auto request = MakeRequest(query);
+    auto const & results = request->Results();
+    size_t constexpr kResultsCount = 3;   // Building should be at the top.
+    TEST_GREATER(results.size(), kResultsCount, ());
+
+    Range const range(results, 0, kResultsCount);
+    HasAddress(range, {}, {});
+  }
+}
+
+UNIT_CLASS_TEST(MwmTestsFixture, Hawaii_Address)
+{
+  // Honolulu
+  ms::LatLon const center(21.3045470, -157.8556760);
```

Comment:
```suggestion
  ms::LatLon const honolulu(21.3045470, -157.8556760);
```

И тогда комментарий не нужен.

Reply:
Мне нравится именно так, как по шаблону с др тестами.
В первую очередь это центр поиска.